padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,195 @@
+"""
+A very simple training example for a Variational Autoencoder(VAE)
+using the mnist database, sacred and the padertorch trainer
+
+Use the following command:
+    python mnist_vae_example.py with logfilepath=/path/to/dir/ epochs=500
+
+Other options for sacred are available, refer to the config() function.
+"""
+import os
+
+import padertorch
+import padertorch.train.optimizer as pt_opt
+
+import sacred
+
+import torch
+import torch.nn as nn
+import torchvision
+import torchvision.transforms as transforms
+
+from einops import rearrange
+
+ex = sacred.Experiment()
+
+
+def get_dataset(batch_size=128):
+    """
+    load MNIST dataset and store it in ./data
+
+    Args:
+        batch_size (int):
+
+    Returns:
+        trainloader and testloader
+
+    """
+    tf = transforms.Compose([transforms.ToTensor()])
+
+    # Get Datasets
+    trainset = torchvision.datasets.MNIST(root='./data', train=True,
+                                          download=True, transform=tf)
+    trainloader = torch.utils.data.DataLoader(trainset, batch_size=batch_size,
+                                              shuffle=True, num_workers=2)
+
+    testset = torchvision.datasets.MNIST(root='./data', train=False,
+                                         download=True, transform=tf)
+    testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size,
+                                             shuffle=False, num_workers=2)
+
```

Comment:
I think we should at least entertain the idea of using lazy_dataset here instead of the DataLoader because this allows for much more freedom in preprocessing and it does not lead to discarding examples if len(dataset)%batch_size !=0

In padertorch.testing.test_db there is already a database object for MNIST which could be used here.